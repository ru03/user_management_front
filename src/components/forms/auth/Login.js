import React from 'react';
import { Box, Button, Card, CardContent, CardActions, CardHeader, Grid, makeStyles, TextField } from '@material-ui/core';

const useStyles = makeStyles(theme => ({
  header: {
    textAlign: 'center',
    spacing: 10,
  },
  card: {
    width: '30%',
  }
}));
const Login = ({ errors, handleChange, handleSubmit, isSubmitting, touched, values }) => {
  const classes = useStyles();
  return (
    <form onSubmit={handleSubmit} data-testid="loginForm">
      <Box mt="25%">
        <Grid
          container
          direction="column"
          justify="center"
          alignItems="center"
        >
          <Card className={classes.card}>
            <CardHeader title="Login" className={classes.header} />
            <CardContent>
              <Grid lg={12} item>
                <TextField
                  id="email"
                  name="email"
                  label="Email"
                  placeholder="email"
                  fullWidth
                  onChange={handleChange}
                  value={values.email}
                  error={touched.email && errors.email ? true : false}
                  helperText={touched.email && errors.email}
                />
              </Grid>
              <Grid lg={12} item>
                <TextField
                  id="password"
                  name="password"
                  label="Password"
                  type="password"
                  placeholder="password"
                  fullWidth
                  onChange={handleChange}
                  value={values.password}
                  error={touched.password && errors.password ? true : false}
                  helperText={touched.password && errors.password}
                />
              </Grid>
            </CardContent>
            <CardActions>
              <Grid lg={12} item>
                <Button
                  id="login_button"
                  data-testid="buttonSubmit"
                  color="primary"
                  disabled={isSubmitting}
                  variant="contained"
                  type="submit"
                  fullWidth
                >
                  Login
                </Button>
              </Grid>
            </CardActions>
          </Card>
        </Grid>
      </Box>
    </form>
  )
}

export default Login;