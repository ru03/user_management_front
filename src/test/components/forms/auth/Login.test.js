import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Login from '../../../../components/forms/auth/Login';

describe('Login Form Component', () => {
  let props;
  beforeEach(() => {
    props = {
      errors: {},
      handleChange: jest.fn(),
      handleSubmit: jest.fn(),
      isSubmitting: false,
      touched: {},
      values: {},
    }
  });

  it('mounts', () => {
    const { asFragment } = render(<Login {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('calls email handle change', () => {
    const { getByPlaceholderText } = render(<Login {...props} />);
    fireEvent.change(getByPlaceholderText('email'), { target: { value: 'value' } });
    expect(props.handleChange).toHaveBeenCalledTimes(1);
    expect(getByPlaceholderText('email').value).toBe('value');
  });

  it('shows error email', () => {
    props = { ...props, errors: { email: 'Email is required!' }, touched: { email: true } }
    const { getByText } = render(<Login {...props} />)
    expect(getByText('Email is required!')).toBeInTheDocument();
  });

  it('calls password handle change', () => {
    const { getByPlaceholderText } = render(<Login {...props} />)
    fireEvent.change(getByPlaceholderText('password'), { target: { value: 'password value' } });
    expect(props.handleChange).toHaveBeenCalledTimes(1);
    expect(getByPlaceholderText('password').value).toBe('password value');
  });

  it('calls handleSubmit when click on Login', () => {
    const { getByTestId } = render(<Login {...props} />)
    fireEvent.submit(getByTestId('loginForm'));
    expect(props.handleSubmit).toHaveBeenCalledTimes(1);
  });

  it('disables submit button if is submitting', () => {
    props = { ...props, isSubmitting: true };
    const { getByTestId } = render(<Login {...props} />)
    expect(getByTestId('buttonSubmit')).toBeDisabled();
  });
});