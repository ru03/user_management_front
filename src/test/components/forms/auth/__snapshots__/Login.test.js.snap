// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Form Component mounts 1`] = `
<DocumentFragment>
  <form
    data-testid="loginForm"
  >
    <div
      class="MuiBox-root MuiBox-root-4"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
      >
        <div
          class="MuiPaper-root MuiCard-root makeStyles-card-2 MuiPaper-elevation1 MuiPaper-rounded"
        >
          <div
            class="MuiCardHeader-root makeStyles-header-1"
          >
            <div
              class="MuiCardHeader-content"
            >
              <span
                class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
              >
                Login
              </span>
            </div>
          </div>
          <div
            class="MuiCardContent-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-lg-12"
            >
              <div
                class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                  data-shrink="false"
                  for="email"
                  id="email-label"
                >
                  Email
                </label>
                <div
                  class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiInput-input"
                    id="email"
                    name="email"
                    placeholder="email"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-lg-12"
            >
              <div
                class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                  data-shrink="false"
                  for="password"
                  id="password-label"
                >
                  Password
                </label>
                <div
                  class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiInput-input"
                    id="password"
                    name="password"
                    placeholder="password"
                    type="password"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiCardActions-root MuiCardActions-spacing"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-lg-12"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
                data-testid="buttonSubmit"
                id="login_button"
                tabindex="0"
                type="submit"
              >
                <span
                  class="MuiButton-label"
                >
                  Login
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </form>
</DocumentFragment>
`;
